Passing pre_save_hook as importstring

## Description

Passing a `pre_save_hook` as an importstring e.g.
```
--FileContentsManager.pre_save_hook='mymodule.foo_function`
```

fails with the following error when launching the Jupyter server:
<details><summary>Troubleshoot Output</summary>
<pre>
    @validate("pre_save_hook")
    def _validate_pre_save_hook(self, proposal):
        value = proposal["value"]
        if isinstance(value, str):
            value = import_item(self.pre_save_hook)
        if not callable(value):
            raise TraitError("pre_save_hook must be callable")
        if self.pre_save_hook is not None:
            warnings.warn(
>               f"Overriding existing pre_save_hook ({self.pre_save_hook.__name__}) with a new one ({value.__name__}).",
                stacklevel=2,
            )
E           AttributeError: 'str' object has no attribute '__name__'
</pre>
</details>


If you edit `jupyter_server_config.py` to import the function, then setting the value is fine.

## Reproduce

Pass `pre_save_hook` or `post_save_hook` as a importstring.

## Expected behavior

Passing as a string or function works equally well.

## Discussion

The problem seems to be because the code assumes that `self.pre_save_hook` is always a function and has the `__name__` dunder.

--- a/jupyter_server/services/contents/manager.py
+++ b/jupyter_server/services/contents/manager.py
@@ -127,7 +127,7 @@ class ContentsManager(LoggingConfigurable):
             value = import_item(self.pre_save_hook)
         if not callable(value):
             raise TraitError("pre_save_hook must be callable")
-        if self.pre_save_hook is not None:
+        if callable(self.pre_save_hook):
             warnings.warn(
                 f"Overriding existing pre_save_hook ({self.pre_save_hook.__name__}) with a new one ({value.__name__}).",
                 stacklevel=2,
@@ -162,7 +162,7 @@ class ContentsManager(LoggingConfigurable):
             value = import_item(value)
         if not callable(value):
             raise TraitError("post_save_hook must be callable")
-        if self.post_save_hook is not None:
+        if callable(self.post_save_hook):
             warnings.warn(
                 f"Overriding existing post_save_hook ({self.post_save_hook.__name__}) with a new one ({value.__name__}).",
                 stacklevel=2,

--- a/tests/services/contents/test_config.py
+++ b/tests/services/contents/test_config.py
@@ -18,6 +18,30 @@ def test_config_did_something(jp_server_config, jp_serverapp):
     )
 
 
+def example_pre_save_hook():
+    pass
+
+
+def example_post_save_hook():
+    pass
+
+
+@pytest.mark.parametrize(
+    "jp_server_config",
+    [
+        {
+            "ContentsManager": {
+                "pre_save_hook": "tests.services.contents.test_config.example_pre_save_hook",
+                "post_save_hook": "tests.services.contents.test_config.example_post_save_hook",
+            },
+        }
+    ],
+)
+def test_pre_post_save_hook_config(jp_serverapp, jp_server_config):
+    assert jp_serverapp.contents_manager.pre_save_hook.__name__ == "example_pre_save_hook"
+    assert jp_serverapp.contents_manager.post_save_hook.__name__ == "example_post_save_hook"
+
+
 async def test_async_contents_manager(jp_configurable_serverapp):
     config = {"ContentsManager": {"checkpoints_class": AsyncCheckpoints}}
     argv = [

